### ISSUE TITLE
Using argparse.Namespace under certain conditions causes E1101(no-member), but for str

### ISSUE THREAD

**Issue by lbenezriravin** [PyCQA/pylint#2428]
### Steps to reproduce
Run pylint on this code:
```
# pylint: disable=missing-docstring, invalid-name, blacklisted-name, unused-argument, unused-variable, too-many-function-args
import argparse


def create_parser_a(foo, bar):
    parser = argparse.ArgumentParser()

    parser.add_argument('my_key')

    args, _ = parser.parse_known_args()
    return args

def create_parser_b(foo):
    parser = argparse.ArgumentParser()

    parser.add_argument('my_key')

    args, _ = parser.parse_known_args()
    return args

def create_parser_c(foo, bar):
    parser = argparse.ArgumentParser()

    parser.add_argument('my_key')

    args = parser.parse_args()
    return args


my_ns_a = create_parser_a('foo', 'bar')
print(my_ns_a.my_key)

my_ns_b = create_parser_b('foo')
print(my_ns_b.my_key)

my_ns_c = create_parser_c('foo', 'bar')
print(my_ns_c.my_key)

my_ns_d = create_parser_b('foo', 'bar')
print(my_ns_d.my_key)
```

### Current behavior
```
$ pipenv run pylint example.py 
************* Module example
example.py:31:6: E1101: Instance of 'str' has no 'my_key' member (no-member)
example.py:40:6: E1101: Instance of 'str' has no 'my_key' member (no-member)
```

### Expected behavior
E1101 should not be raised in any instance, but especially not for `'str'`
What baffles me is that, if I have fewer than two arguments to my namespace-creating function, or I use `parse_args` instead of `parse_known_args`, the problem goes away. The problem returns if I *call* the one-arg function with two arguments, as well.

This might be related to #2413, but seemed interesting and distinct enough to merit its own issue

possible also related to #2425 

### pylint --version output
```
pylint 2.1.1
astroid 2.0.4
Python 3.7.0 (default, Jun 28 2018, 08:59:52) 
[GCC 7.3.1 20180303 (Red Hat 7.3.1-5)]
```

**Comment by PCManticore** [comment]
Thanks for creating an issue! It's completely possible that `pylint` infers the result of `parse_known_args` as a string instead of a namespace object.

**Comment by lbenezriravin** [comment]
I'd be glad if it were that simple, but I'm not so sure -- I can't reproduce this bug by calling `parse_known_args` directly. The following raises nothing:
```
parser = argparse.ArgumentParser()
parser.add_argument('my_key')
args, _ = parser.parse_known_args()
print(args.my_key)
```
I have to return the namespace given in the first part of the tuple as part of some other function.

**Comment by PCManticore** [comment]
Awesome, thanks for double checking that's not the case @lbenezriravin

**Comment by jacobtylerwalls** [comment]
Can't reproduce on pylint 2.14.3
